text/0005-config_design.md

```diff
@@ -0,0 +1,110 @@
+- Feature Name: Redesign of nushells config structure
+- Start Date: (fill me in with today's date, 2021-03-18)
+- RFC PR: [nushell/rfcs#0005](https://github.com/nushell/rfcs/pull/0005)
+
+# Summary
+[summary]: #summary
+
+This RFC shall provide the basis for a structured discussion, how the nushell config could be improved.
+
+# Motivation
+[motivation]: #motivation
+Important: This document is written as if, https://github.com/nushell/nushell/pull/3041 would already be accepted and merged. The PR is currently in review.
+
+
+Nushell evolved over time and features got added. The current config format may need to be adapted to better represent nushells capabilities to new users.
+
+The config structure is mostly well done, but some enhancements I can think of:
+1. Every `[...]` table has a name without config, but there is `[color_config]` which I think should be renamed to `theme` (better consistency, better naming).
```

Comment:
i'm not sure i agree about `[color_config]` being named `[theme]` because `theme` infers that there are named themes which we currently don't support, but I, for one, would definitely like to support named themes. Before that happens the coloring system needs an overhaul. there's a pr where i implemented a change that would respect the `LS_COLORS` env var but it seemed to really slow down the system. it also stopped one `ls` parameter from working. neither issue should be the case.

Reply:
Good point. What about `[colors]`?

Reply:
`[table_colors]`?

Reply:
Deal!